### ISSUE TITLE
.NET 6.0.3 fails when installing from Windows Update 

### ISSUE THREAD

**Issue by pedoc** [dotnet/sdk#24294]
### Describe the bug
Installing .NET 6.0.3 from Windows Update fails with error code 0x80070643,but it can be installed normally using the stand-alone installer .

![image](https://user-images.githubusercontent.com/6604230/157363205-b8aad7ca-17e4-4c76-9ee6-cdd2a7012a49.png)

### Further technical details
- Include the output of `dotnet --info`
```
.NET SDK (反映任何 global.json):
 Version:   7.0.100-preview.3.22157.4
 Commit:    b34f562bf5

运行时环境:
 OS Name:     Windows
 OS Version:  10.0.22000
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\7.0.100-preview.3.22157.4\

Host (useful for support):
  Version: 7.0.0-preview.3.22151.6
  Commit:  2330b4bef6

.NET SDKs installed:
  2.1.818 [C:\Program Files\dotnet\sdk]
  3.1.417 [C:\Program Files\dotnet\sdk]
  5.0.406 [C:\Program Files\dotnet\sdk]
  6.0.200 [C:\Program Files\dotnet\sdk]
  7.0.100-preview.3.22157.4 [C:\Program Files\dotnet\sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.All 2.1.30 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.App 2.1.30 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.23 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 5.0.15 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 6.0.2 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 7.0.0-preview.3.22152.6 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 2.1.30 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.23 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 5.0.15 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.2 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 7.0.0-preview.3.22151.6 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 3.1.23 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 5.0.15 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 6.0.2 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 7.0.0-preview.3.22128.8 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]

To install additional .NET runtimes or SDKs:
  https://aka.ms/dotnet-download
```
- envs
1. windows 11 22000.556 x64

**Comment by marcpopMSFT** [comment]
@pedoc do you have Visual Studio installed?  CC @joeloff

**Comment by joeloff** [comment]
@csnellgrove1-zz as an FYI

@pedoc, can you run `Get-WindowsUpdateLog` from powershell and attach the log? The cmdlet will merge the Microsoft Update ETL files into a text file and drop it on your desktop.

**Comment by pedoc** [comment]
@marcpopMSFT yep,I can update normally from Visual Studio Installer.

PS:List of Visual Studio Versions 
![image](https://user-images.githubusercontent.com/6604230/157580662-5f7d7aaa-71b7-4ff7-b762-43b8a82bd7cf.png)

**Comment by pedoc** [comment]
@joeloff

When I tried to update from Windows Update again today, it still failed, here is the update log.

[WindowsUpdate.log](https://github.com/dotnet/sdk/files/8219916/WindowsUpdate.log)

![image](https://user-images.githubusercontent.com/6604230/157580848-e0855f58-5016-4293-af2a-123eb4386e0b.png)

**Comment by EamonNerbonne** [comment]
I seem to have the same issue.  

I already installed 6.0.3 manually, and VS 17.1.1, and later windows update asked to install KB5012417, and fails with 0x80070643.  I'm running Windows 11 Pro, and on the OS build as @pedoc - 22000.556.  I'm running US-english language, region Netherlands, Regional format English (Canada), in case that matters (I expect not?)

I've attached my update log as per the instructions provided above; perhaps there's some helpful similarity somewhere.

[WindowsUpdate.log](https://github.com/dotnet/sdk/files/8221404/WindowsUpdate.log)

**Comment by jmelhus** [comment]
Same issue here.

Windows 11 x64, 22000.556

dotnet --info:
```
.NET SDK (reflecting any global.json):
 Version:   6.0.101
 Commit:    ef49f6213a

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.22000
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\6.0.101\

Host (useful for support):
  Version: 6.0.1
  Commit:  3a25a7f1cc

.NET SDKs installed:
  3.1.417 [C:\Program Files\dotnet\sdk]
  5.0.104 [C:\Program Files\dotnet\sdk]
  5.0.202 [C:\Program Files\dotnet\sdk]
  5.0.212 [C:\Program Files\dotnet\sdk]
  6.0.101 [C:\Program Files\dotnet\sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.All 2.1.27 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.1.30 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.App 2.1.27 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.1.30 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.14 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.23 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 5.0.4 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 5.0.5 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 5.0.15 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 6.0.1 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 2.1.27 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.1.30 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.14 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.23 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 5.0.4 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 5.0.5 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 5.0.15 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.1 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 3.1.14 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 3.1.23 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 5.0.4 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 5.0.5 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 5.0.15 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 6.0.1 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
```

**Comment by EamonNerbonne** [comment]
Update: after around a week or so, the update is no longer offered.  I don't think it's in the list of installed updates either; it's just gone (which is fine, I believe).

The new "Update History" UI in win11 is incidentally really unusable for something like this. It looks simpler, but since it's grouped by type of update, with no way to add extra columns nor filter or sort by a choice of attribute, it's quite a lot of clicking to be sure something is *not* there.  A plain grid view would have been much more convenient.

**Comment by jmelhus** [comment]
Same here, I updated Visual Studio where 6.0.3. was included. After that Windows update doesn't try to update anymore (and the error above is gone).

**Comment by pedoc** [comment]
Yes, this issue seems to be gone, should I close it?

**Comment by marcpopMSFT** [comment]
Per comments, I'm closing this as we fixed the MU offering for 6.0.3 we believe.

**Comment by suhailzone** [comment]
> Same here, I updated Visual Studio where 6.0.3. was included. After that Windows update doesn't try to update anymore (and the error above is gone).

To resolve that issue, which exact version of Visual Studio do I need to install?. any download link.
